lib/app/features/ion_connect/providers/ion_connect_upload_notifier.m.dart

```diff
@@ -170,13 +179,77 @@ class IonConnectUploadNotifier extends _$IonConnectUploadNotifier {
           UploadResponse.fromJson(json.decode(response.data as String) as Map<String, dynamic>);
 
       if (uploadResponse.status != 'success') {
+        // LOGGER: HTTP logging - upload_result_err
+        _logHttpUploadResultErr(
+          host: host,
+          response: response,
+          msg: uploadResponse.message,
+        );
         throw Exception(uploadResponse.message);
       }
+
+      // LOGGER: HTTP logging - upload_result_ok
+      _logHttpUploadResultOk(
+        host: host,
+        response: response,
+      );
+
       return uploadResponse;
     } catch (error) {
       throw FileUploadException(error, url: url);
     }
   }
+
+  // Large multipart uploader that generates its own token
+  Future<UploadResponse> _uploadLargeMultipart({
+    required String url,
+    required MediaFile file,
+    required Uint8List fileBytes,
+    String? alt,
+    CancelToken? cancelToken,
+    EventSigner? customEventSigner,
+  }) async {
+    // Generate auth token with conditional 10100 attestation
+    final authToken = await generateAuthorizationToken(
+      ref: ref,
+      url: url,
+      method: 'POST',
+      fileBytes: fileBytes,
+      customEventSigner: customEventSigner,
+    );
+
+    final dio = ref.read(dioHttp2Provider);
+    final feedConfig = await ref.watch(feedConfigProvider.future);
+
+    return LargeMediaUploadService(
+      dio: dio,
+      maxConcurrentPartials: feedConfig.concurrentBigFileUploadChunks,
+    ).upload(
+      url: url,
+      file: file,
+      authToken: authToken,
+      alt: alt,
+      fileBytes: fileBytes,
+      cancelToken: cancelToken,
+    );
+  }
+
+  /// Minimal helper to check if the error is related to on-behalf/attestation errors
+  /// Later move to a more generic helper!!!
+  static bool _isOnBehalfAttestationError(Object error) {
+    if (error is FileUploadException) {
+      final message = error.message.toLowerCase();
+
+      final hasAuthCode = message.contains('status code of 403') ||
+          message.contains(' 403 ') ||
```

Comment:
maybe it would be more reliable to save the source `error` in the `FileUploadException` class field, then here check that that attached `error is DioException && (error.response.statusCode == HttpStatus.unauthorized || error.statusCode == HttpStatus.forbidden)` + that phrase in the message?